src/toil/batchSystems/abstractBatchSystem.py

```diff
@@ -165,6 +166,79 @@ def workerCleanup(workerCleanupInfo):
+        :return: Amount of disk used by non-Toil purposes
+        '''
+        # Get the statistics of the device hosting workDir
+        workDir = workDir or os.getenv('TOIL_WORKDIR')
+        if workDir:
+            toilDiskUsed = AbstractBatchSystem.folderWalk(workDir)
+            # Directory that os.statvfs should be run on.  If workDir is known, use it. Else use the
+            # directory where the workdir will be placed when it is set up.
+            dirToStat = workDir
+        else:
+            toilDiskUsed = 0
+            dirToStat = tempfile.gettempdir()
+        diskStats = os.statvfs(dirToStat)
+        # Total Disk used = block size * (total blocks - free blocks)
+        diskUsed = diskStats.f_frsize * (diskStats.f_blocks - diskStats.f_bfree)
+        # Disk used for non-toil purposes = Total disk used - Disk used by toil (workDir)
+        nonToilDiskUsed = diskUsed - toilDiskUsed
+        return nonToilDiskUsed
+
+    @staticmethod
+    def folderWalk(workDir):
+        '''
+        This job will walk through workDir and return the cumulative filesize in bytes of all the
+        files in the directory and its subdirectories.
+        :param workDir: path to the workflow working directory
+        :return: cumulative size in bytes of all files in the directory
+        '''
+        total_size = 0
+        for dirpath, dirnames, filenames in os.walk(workDir):
+            # The value from running stat on each linked file is equal. To prevent the same file
+            # from ebing counted multiple times, we count the value only once in the cache dir. This
+            # assumes that the user never creates a hard link to/from a location in the workdir.
+            considerLinked = 'cache' in dirpath
+            total_size += AbstractBatchSystem._getFolderSize(dirpath, filenames,
+                                                             considerLinked=considerLinked)
+        return total_size
+
+    @staticmethod
+    def _getFolderSize(dirpath, filenames,considerLinked=False):
+        '''
+        aux for folderWalk
+        :param dirpath: Dir path
+        :param filenames: Files in the directory
+        :param considerLinked: Should we count the sizes of hard linked files too?
+        :return:
+        '''
+        folderSize = 0
+        for f in filenames:
+            fp = os.path.join(dirpath, f)
+            fileStats = os.stat(fp)
+            if fileStats.st_nlink > 1:
+                folderSize += fileStats.st_size if considerLinked else 0
+            else:
+                folderSize += fileStats.st_size
+        return folderSize
+
+    @staticmethod
+    def getFileSystemSize(workDir):
+        '''
+        Returns the size of the file system hosting workDir, in bytes
+        :param workDir: path to workDir. Can be None!
+        :return: size of the file system
+        '''
+        dirToStat = workDir or os.getenv('TOIL_WORKDIR') or tempfile.gettempdir()
```

Comment:
Do we really need all these fallbacks? And if so we should encapsulate in a function.